kafka.go

```diff
@@ -123,13 +141,13 @@ func (client *KafkaClient) Publish(publishBuilder *PublishBuilder) error {
 		topic := pubTopic
 		go func() {
 			err = backoff.RetryNotify(func() error {
-				return client.publishEvent(publishBuilder.ctx, topic, config, message)
+				return client.publishEvent(publishBuilder.ctx, topic, publishBuilder.eventName, config, message)
 			}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxBackOffCount),
 				func(err error, _ time.Duration) {
-					logrus.Debugf("retrying publish event: error %v : ", err)
+					log.Debugf("retrying publish event: error %v: ", err)
 				})
 			if err != nil {
-				logrus.Errorf("unable to publish event. topic : %s , event : %s , error : %v", topic,
+				log.Errorf("unable to publish event. topic: %s , event: %s , error: %v", topic,
```

Comment:
this is what I asked before, how would the caller know if the event cannot be published?
as part of resiliency, the event data cannot be lost without conscious decision by the caller

Reply:
do you think we need a channel to send the error to client? but, it will increase complexity in client

Reply:
After had discussion with Yeremia, he has a suggestion to use callback function to handle the event if we got an error and the callback will be supplied by client. what do you think?

Reply:
callback sounds good to give async report

Reply:
Already added the callback func, please help to recheck it
thanks